src/messages.ts

```diff
@@ -22,7 +22,9 @@ function waitForConnectionOpen(this: any) {
 }
 
 export function handleMessage(this: any, msg: { data: string }): void {
-  const { status, reason, event, connectionId } = JSON.parse(msg.data)
+  const { status, reason, event, connectionId, serverVersion } = JSON.parse(
+    msg.data
+  )
```

Comment:
How does error handling work here? What happens if the server sends invalid JSON in msg.data?

Reply:
Yeah it assumes that the server will send valid JSON, otherwise the error will be thrown within the `handleMessage` function just like the other errors, which will only affect the successful delivery of that particular event. 
If we setup some sort of endpoint on our server in the future to report errors within our client libraries, it might be worth adding error handling here so that we can be notified when the server isn't sending valid JSON via WebSockets